protob/messages-binance.proto

```diff
@@ -0,0 +1,166 @@
+syntax = "proto2";
+package hw.trezor.messages.binance;
+
+// Sugar for easier handling in Java
+option java_package = "com.satoshilabs.trezor.lib.protobuf";
+option java_outer_classname = "TrezorMessageBinance";
+
+/**
+ * Request: Ask the device for a Binance address.
+ * @start
+ * @next BinanceAddress
+ * @next Failure
+ */
+message BinanceGetAddress {
+    repeated uint32 address_n = 1;  // BIP-32-style path to derive the key from master node
+    optional bool show_display = 2; // optionally prompt for confirmation on trezor display
+}
+
+/**
+ * Response: A Binance address.
+ * @end
+ */
+message BinanceAddress {
+    optional string address = 1;    // prefixed bech32 Binance address
+}
+
+/**
+ * Request: Ask device for a public key corresponding to address_n path.
+ * @start
+ * @next BinancePublicKey
+ */
+message BinanceGetPublicKey {
+    repeated uint32 address_n = 1;  // BIP-32 path to derive the key from master node
+    optional bool show_display = 2; // optionally show on display before sending the result
+}
+
+/**
+ * Response: A public key corresponding to address_n path.
+ * @end
+ */
+message BinancePublicKey {
+    optional bytes public_key = 1;
+}
+
+/**
+ * Request: Starts the Binance transaction protocol flow.
+ * A transaction consists of these common fields and a series of Binance<Any>Msg messages. 
+ * These parts form a JSON structure (a string) in Trezor's memory, which is signed to produce a BinanceSignedTx.
+ * @start
+ * @next BinanceTxRequest
+ * @next Failure
+*/
+message BinanceSignTx {
+    repeated uint32 address_n = 1; // BIP-32-style path to derive the key from master node
+    optional uint32 msg_count = 2; // count of Binance<Any>Msg to be included in this tx
+    optional sint64 account_number = 3;
+    optional string chain_id = 4;
+    optional string memo = 5;
+    optional sint64 sequence = 6;
+    optional sint64 source = 7;
+}
+
+/**
+ * Response: Trezor requests the next message or signals that it is ready to send a BinanceSignedTx.
+ * @next BinanceTransferMsg
+ * @next BinanceOrderMsg
+ * @next BinanceCancelMsg
+ * @next BinanceSignedTx
```

Comment:
this is wrong, instead, `@next BinanceSignedTx` should be a possibility on each `Binance<Any>Msg`.

(same rules apply to Trezor, it won't be sending two messages in sequence, so after `msg_count` `BinanceTxRequest`s the next response should be `BinanceSignedTx`)